sentry-ruby/lib/sentry-ruby.rb

```diff
@@ -385,6 +385,40 @@ def start_transaction(**options)
       get_current_hub.start_transaction(**options)
     end
 
+    # Records the block's execution as a child of the current span.
+    # If the current scope doesn't have a span, the block would still be executed but the yield param will be nil.
+    # @param attributes [Hash] attributes for the child span.
+    # @yieldparam child_span [Span, nil]
+    # @return yield result
+    #
+    # @example
+    #   Sentry.with_child_span(op: "my operation") do |child_span|
+    #     child_span.set_data(operation_data)
+    #     child_span.set_description(operation_detail)
+    #     # result will be returned
+    #   end
+    #
+    def with_child_span(**attributes, &block)
+      current_span = get_current_scope.get_span
+
+      if current_span
+        result = nil
+
+        begin
+          current_span.with_child_span(**attributes) do |child_span|
+            get_current_scope.set_span(child_span)
```

Comment:
Just an observation, all the current usages of `set_span` set it to the transaction and we don't really use `scope#get_span` either so this will be the first time the span on the scope gets used right?

```
➜  sentry-ruby git:(master) ✗ ag --ignore 'spec' set_span
sentry-rails/lib/sentry/rails/action_cable.rb
18:              scope.set_span(transaction) if transaction

sentry-rails/lib/sentry/rails/active_job.rb
31:                scope.set_span(transaction) if transaction

sentry-resque/lib/sentry/resque.rb
28:              scope.set_span(transaction) if transaction

sentry-delayed_job/lib/sentry/delayed_job/plugin.rb
22:            scope.set_span(transaction) if transaction

sentry-sidekiq/lib/sentry/sidekiq/sentry_context_middleware.rb
21:        scope.set_span(transaction) if transaction

sentry-ruby/lib/sentry/scope.rb
132:    def set_span(span)

sentry-ruby/lib/sentry/rack/capture_exceptions.rb
23:            scope.set_span(transaction) if transaction
```

Reply:
Yes, because most the usages in the SDK are to create the top-level transaction. But as described in #1784, `sentry-rails`'s subscriber should use spans instead. And users will generally use `get_span` and `set_span` for spans more than transactions.